streams/jvm/src/main/scala/zio/stream/compression/package.scala

```diff
@@ -0,0 +1,104 @@
+package zio.stream
+
+import java.util.zip.{ DataFormatException, Inflater }
+import java.{ util => ju }
+
+import scala.annotation.tailrec
+
+import zio._
+
+package object compression {
+
+  /**
+   * Decompresses deflated stream. Compression method is described in https://tools.ietf.org/html/rfc1951.
+   *
+   * @param noWrap  Whether is wrapped in ZLIB header and trailer, see https://tools.ietf.org/html/rfc1951.
+   *                For HTTP 'deflate' content-encoding should be false, see https://tools.ietf.org/html/rfc2616.
+   * @param bufferSize Size of buffer used internally, affects performance.
+   **/
+  def inflate(
+    bufferSize: Int = 64 * 1024,
+    noWrap: Boolean = false
+  ): ZTransducer[Any, CompressionException, Byte, Byte] = {
+    def makeInflater(
+      bufferSize: Int
+    ): ZManaged[Any, Nothing, Option[zio.Chunk[Byte]] => ZIO[Any, CompressionException, Chunk[Byte]]] =
+      ZManaged
+        .make(ZIO.effectTotal((new Array[Byte](bufferSize), new Inflater(noWrap)))) {
+          case (_, inflater) => ZIO.effectTotal(inflater.end())
+        }
+        .map {
+          case (buffer, inflater) => {
+            case None =>
+              ZIO.succeed {
```

Comment:
Technically this should be `ZIO.effectTotal` (`ZIO.succeed` is lazy and just forwards to `effectTotal` but it's nicer for maintainability to use `effectTotal`).